api4/user_test.go

```diff
@@ -495,66 +495,116 @@ func TestGetUserByUsername(t *testing.T) {
 func TestGetUserByEmail(t *testing.T) {
 	th := Setup().InitBasic()
 	defer th.TearDown()
-	Client := th.Client
-
-	showEmailAddress := th.App.Config().PrivacySettings.ShowEmailAddress
-	showFullName := th.App.Config().PrivacySettings.ShowFullName
-	defer func() {
-		th.App.UpdateConfig(func(cfg *model.Config) { cfg.PrivacySettings.ShowEmailAddress = showEmailAddress })
-		th.App.UpdateConfig(func(cfg *model.Config) { cfg.PrivacySettings.ShowFullName = showFullName })
-	}()
 
 	user := th.CreateUser()
 
-	ruser, resp := Client.GetUserByEmail(user.Email, "")
-	CheckNoError(t, resp)
-	CheckUserSanitization(t, ruser)
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		cfg.PrivacySettings.ShowEmailAddress = true
+		cfg.PrivacySettings.ShowFullName = true
+	})
 
-	if ruser.Email != user.Email {
-		t.Fatal("emails did not match")
-	}
+	t.Run("should be able to get another user by email", func(t *testing.T) {
+		ruser, resp := th.Client.GetUserByEmail(user.Email, "")
+		CheckNoError(t, resp)
+		CheckUserSanitization(t, ruser)
 
-	ruser, resp = Client.GetUserByEmail(user.Email, resp.Etag)
-	CheckEtag(t, ruser, resp)
+		if ruser.Email != user.Email {
+			t.Fatal("emails did not match")
+		}
+	})
 
-	_, resp = Client.GetUserByEmail(GenerateTestUsername(), "")
-	CheckBadRequestStatus(t, resp)
+	t.Run("should return not modified when provided with a matching etag", func(t *testing.T) {
+		_, resp := th.Client.GetUserByEmail(user.Email, "")
+		CheckNoError(t, resp)
 
-	_, resp = Client.GetUserByEmail(th.GenerateTestEmail(), "")
-	CheckNotFoundStatus(t, resp)
+		ruser, resp := th.Client.GetUserByEmail(user.Email, resp.Etag)
+		CheckEtag(t, ruser, resp)
+	})
 
-	// Check against privacy config settings
-	th.App.UpdateConfig(func(cfg *model.Config) { cfg.PrivacySettings.ShowEmailAddress = false })
-	th.App.UpdateConfig(func(cfg *model.Config) { cfg.PrivacySettings.ShowFullName = false })
+	t.Run("should return bad request when given an invalid email", func(t *testing.T) {
+		_, resp := th.Client.GetUserByEmail(GenerateTestUsername(), "")
+		CheckBadRequestStatus(t, resp)
+	})
 
-	ruser, resp = Client.GetUserByEmail(user.Email, "")
-	CheckNoError(t, resp)
+	t.Run("should return 404 when given a non-existent email", func(t *testing.T) {
+		_, resp := th.Client.GetUserByEmail(th.GenerateTestEmail(), "")
+		CheckNotFoundStatus(t, resp)
+	})
 
-	if ruser.Email != "" {
-		t.Fatal("email should be blank")
-	}
-	if ruser.FirstName != "" {
-		t.Fatal("first name should be blank")
-	}
-	if ruser.LastName != "" {
-		t.Fatal("last name should be blank")
-	}
+	t.Run("should sanitize full name for non-admin based on privacy settings", func(t *testing.T) {
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			cfg.PrivacySettings.ShowFullName = false
```

Comment:
Could you assign both `ShowEmailAddress` and `ShowFullName` explicitly in these blocks, just to make it clear under what conditions the test is operating and not having to rely on inferring from the last block as such?

Reply:
I can do that for those blocks, but I'm on the fence on how to do it otherwise since those other ones still depend on being ran with `ShowEmail` set to true

Reply:
Actually, I just added `ShowEmailAddress = true` to the ones that are testing for full name since the opposite case isn't true (ie the `ShowEmailAddress` tests don't look for the user's full name)